traceutil/roundtrip_test.go

```diff
@@ -0,0 +1,83 @@
+package traceutil
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptrace"
+	"testing"
+)
+
+type roundTripperFunc func(*http.Request) (*http.Response, error)
+
+func (f roundTripperFunc) RoundTrip(req *http.Request) (*http.Response, error) { return f(req) }
+
+func BenchmarkRoundTrip(b *testing.B) {
+	b.Run("basic", func(b *testing.B) {
```

Comment:
I'd prefer to have different Benchmark functions written for basic and parallel.

Sub-tests and benchmarks are useful if you share code in between test cases. In our case, there is no code is shared and having sub-benchs makes the code quite indented.